### PR TITLE
Workaround pip ubuntu bug.

### PR DESCRIPTION
**Pull request by mattmb**
There is a problem in the current ubuntu pip package: https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1306991

This works around it by installing pip from a bash script if the pip command itself is broken.

### CODE REVIEW

**Comment by ashb** [comment]
How much do we trust pypa.io? Would it be worth copying that file into this formula and doing file.managed instead?

Does this download run every single time we highstate? Can we make it conditional somehow so it only runs if the pip is broken?

**Comment by mattmb** [comment]
I did make it conditional :smile:

**Comment by mattmb** [comment]
But yeah maybe we should file.managed it, that's a good shout.

**Comment by mattmb** [comment]
@ashb updated to manage the file

**Comment by ashb** [comment]
Aaaand changelog please :D

**Comment by mattmb** [comment]
Good point! added.

**Comment by mattmb** [comment]
This doesn't seem to work on first run...

**Comment by mattmb** [comment]
OK so it breaks once the requests module gets installed. So to fix this properly I think we should do pip.installed with the requests module, then the fix state.

**Comment by mattmb** [comment]
Had to use a cmd because the pip.installed state would have failed but now happy that this works on first run and subsequent highstates.

File: python/init.sls [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+      - cmd: requests
+  file.managed:
+    - name: /tmp/get-pip.py
+    - mode: 700
+    - source: salt://python/files/get-pip.py
+
+requests:
+  cmd.run:
+    - name: pip install requests==2.5.3
+    - unless: pip list | grep "requests (2.5.3)"
```
Comment (ashb) [inline python/init.sls:22]: We probably need a comment saying why we're using a cmd.run rather than the pip state here

**Comment by mattmb** [comment]
@ashb added a comment

**Comment by lukaszraczylo** [comment]
LGTM.

**Pull request merged**